### PR TITLE
Fix preview not updating when autosave is off - issue 120

### PR DESCRIPTION
**Pull request by dehru**
### What does this PR do?

This PR address this issue that was introduced in newly published version 50.14.0.  The issue is captured here. https://github.com/forcedotcom/soql-tooling/issues/120

### What issues does this PR fix or reference?
https://github.com/forcedotcom/soql-tooling/issues/120

### CODE REVIEW

File: packages/soql-builder-ui/src/modules/querybuilder/services/toolingModelService.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ export interface ToolingModel extends IMap {
   limit: string;
   errors: List<Map>;
   unsupported: List<Map>;
+  originalSoqlStatement: string;
```
Comment (dehru) [inline packages/soql-builder-ui/src/modules/querybuilder/services/toolingModelService.ts:31]: The preview component is reactive to this property.  Previously, we were only displaying the soql query from the text document when we got a onDidChangeTextDocument event.  But when the user interacts with the UI, we also generate the soql statement.  So we can use the soql statement generated by the user interaction with the form to also display the soql statement in the preview by adding this property to the immutable model.

File: packages/soql-builder-ui/package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@salesforce/soql-builder-ui",
   "description": "SOQL Builder UI with LWC",
-  "version": "0.0.28",
+  "version": "0.0.29",
```
Comment (dehru) [inline packages/soql-builder-ui/package.json:4]: ready to publish a new version once this is QAd

**Review by jonnyhork — APPROVED** [review]
Nice, this looks good. I can't remember why we didn't do it this way before? Also, do we know the source of the bug, when and what changed?

**Comment by dehru** [comment]
> Nice, this looks good. I can't remember why we didn't do it this way before? Also, do we know the source of the bug, when and what changed?

So we used to always show the preview from the soql statement that was generated by the UI.  But then we had that issue where the user could open a soql query from a text editor that we didn't support...but we still wanted to show them the soql statement and allow them to run what they see.  So we switched it to only update the preview from the incoming soql statement.

I think what we want is show the soql statement from the text editor when the change event happens.  But also, when the user interacts with our form, immediately update the soql statement.  Since we are sending that to the text editor anyway.

It's like the best of both worlds now.  I hope.

**Comment by jonnyhork** [comment]
QA Passed for me

**Pull request merged**